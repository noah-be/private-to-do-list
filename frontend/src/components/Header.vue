<script setup>
import Navbar from "@/components/Navbar.vue";
</script>

<template>
  <header class="header">
    <h1>Task Master</h1>
    <Navbar />
    <button id="profile-button" class="user-profile-icon">
      <img src="/images/profile-user.png" alt="User Profile" />
    </button>
  </header>
</template>

<style lang="scss">
// header {
//   background-color: $primary-color;
//   color: $secondary-color;
//   display: flex;
//   justify-content: space-between;
//   align-items: center;
//   height: 50px;
//   padding: 10px;

//   .user-profile-icon img {
//     display: block;
//     width: 100%;
//     height: auto;
//   }

//   #profile-button {
//     width: 50px;
//     height: 50px;
//   }
// }
</style>
