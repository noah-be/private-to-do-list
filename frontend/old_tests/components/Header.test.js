import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Header from "@/components/Header.vue";

vi.mock("@/components/Navbar.vue", () => ({
  default: {
    template: "<nav></nav>",
  },
}));

describe("Header.vue", () => {
  it("renders correctly and matches snapshot", () => {
    const wrapper = mount(Header);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
