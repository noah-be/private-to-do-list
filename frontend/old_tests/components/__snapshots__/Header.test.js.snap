// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue > renders correctly and matches snapshot 1`] = `
"<v-app-bar app="" dense="" color="primary" dark="">
  <v-container>
    <v-row align="center" justify="space-between">
      <v-col cols="auto">
        <h1 class="text-h5 text-white">Task Master</h1>
      </v-col>
      <v-col cols="auto">
        <nav></nav>
      </v-col>
      <v-col cols="auto">
        <v-btn icon="">
          <v-img src="/images/profile-user.png" alt="User Profile" width="40" height="40" cover="" class="rounded-circle"></v-img>
        </v-btn>
      </v-col>
    </v-row>
  </v-container>
</v-app-bar>"
`;
